c/dataservice.c

```diff
@@ -107,18 +107,35 @@ static void *lookupDLLEntryPoint(char *libraryName, char *functionName){
     | RTLD_NOW
 #endif
     ;
-  void *library = dlopen(libraryName,flags);
-  if (library){
-    printf("dll handle for %s = 0x%" PRIxPTR "\n",libraryName, library);
-    ep = dlsym(library,functionName);
-    /* dlclose(library); - this will really close the DLL, like not be able to call */
-    if (ep == NULL){
-      printf("%s.%s could not be found -  dlsym error %s\n", libraryName, functionName, dlerror());
+  FileInfo info;
+  int status = 0;
+  int returnCode = 0;
+  int reasonCode = 0;
+  status = fileInfo(libraryName, &info, &returnCode, &reasonCode);
+  if (status == 0) {
+    if (!(info.attributeFlags & 0x2)) {
```

Comment:
I suggest the following changes:
* Use a ```#define``` instead of a magic number:
```diff
diff --git a/h/unixfile.h b/h/unixfile.h
index 581ba89..e6e9d98 100644
--- a/h/unixfile.h
+++ b/h/unixfile.h
@@ -164,7 +164,13 @@ typedef struct BPXYSTAT_tag{
   short    fileTaggingTags;
   int      reserved2[2];
   int64    blocks;
-  int      attributeFlags;
+  uint32_t reservedAttributeFlags: 24;
+  uint8_t  attributeFlags;
+#define BPXYSTAT_ATTR_SHARELIB    0x10
+#define BPXYSTAT_ATTR_NOSHAREAS   0x08
+#define BPXYSTAT_ATTR_APFAUTH     0x04
+#define BPXYSTAT_ATTR_PROGCTL     0x02
+#define BPXYSTAT_ATTR_EXTLINK     0x01
   /* 0x10 shared lib
      0x08 no shareas
      0x04 APF
```
In ```daraservice.c```:
```c
     if (!(info.attributeFlags & BPXYSTAT_ATTR_PROGCTL)) { 
```
* Don't use your TSO name, squash the commits